contracts/inflow/src/contract.rs

```diff
@@ -486,3 +1171,151 @@ fn create_set_denom_metadata_msg(
         ),
     })
 }
+
+/// Updates the list of user's pending withdrawal requests by adding and/or removing specified withdrawal IDs.
+fn update_user_withdrawal_requests_info(
+    storage: &mut dyn Storage,
+    user: Addr,
+    config: &Config,
+    withdrawals_to_add: Option<Vec<u64>>,
+    withdrawals_to_remove: Option<Vec<u64>>,
+    should_check_withdrawal_limit: bool,
```

Comment:
I added this check to be optional for the following reason: If user already has maximum allowed withdrawal requests (let's say 10) and we `update_config()` to 5 afterwards, we wouldn't be able to execute `cancel_withdrawal()` and `claim_unbonded_withdrawals()` for this user, if we try to remove less than 5 of its withdrawal requests. It is a bit far fetched, but better to be safe than sorry.